test/tests/kernels/ad_value/tests

```diff
@@ -7,4 +7,12 @@
     design = "syntax/Kernels/index.md"
     issues = "#5658"
   [../]
+  [./generic_test]
+    type = 'Exodiff'
+    input = 'generic_value.i'
+    exodiff = 'generic_value_out.e'
+    requirement = "GenericKernel will act as switch allowing inheritance from proper kernel depending on boolean is_ad"
```

Comment:
We want to keep implementation details out of our requirements. The requirement should drive the design, not visa versa. Want something more along the lines of "The system shall provide a generic kernel object that allows for easy switching between hand-coded and automatic computation of the Jacobian, and encourages maximum kernel code-reuse."